src/controlplane/scraper.go

```diff
@@ -99,8 +101,10 @@ func NewScraper(config *config.Config, providers Providers, options ...ScraperOp
 		return nil, fmt.Errorf("fetching K8s version: %w", err)
 	}
 
-	// Building pod lister and closers for pod autodisover.
-	s.buildLister()
+	// Building pod and secret lister and closers.
+	s.buildListers()
```

Comment:
This still confuses me a bit. I do not like that it saves the lister into the object in a hidden way, moreover the client doesn't need to save those object, since it merely passes them two lines below

Reply:
(at least the secret lister), wouldn't be better to return the two and saving explicitely the podLister and passing by the secret one without saving it into the struct?